knit/core.py

```diff
@@ -87,6 +90,7 @@ class Knit(object):
 
     JAR_FILE = "knit-1.0-SNAPSHOT.jar"
     JAVA_APP = "io.continuum.knit.Client"
+    instances = weakref.WeakSet()
```

Comment:
I'd probably move the `instances` set to a global, and the `cleanup` class method to a function, but that's just a style thing. If left on the class I think they both should be private (`_` prefixed) at least. Otherwise this looks good.